src/System.CommandLine.Suggest/FileSuggestionRegistration.cs

```diff
@@ -52,9 +53,18 @@ public Registration FindRegistration(FileInfo soughtExecutable)
                 return null;
             }
 
-            var completionTarget =
-                File.ReadAllLines(_registrationConfigurationFilePath).LastOrDefault(line =>
-                                                                                        line.StartsWith(soughtExecutable.FullName, StringComparison.OrdinalIgnoreCase));
+            string completionTarget = null;
+            using (var sr = new StreamReader(_registrationConfigurationFilePath, Encoding.UTF8))
+            {
+                string? line;
+                while ((line = sr.ReadLine()) != null)
```

Comment:
Or maybe do away with the separate variable declaration and do this?

```suggestion
                while (sr.ReadLine() is { } line)
```

Reply:
I was under the impressions that this was targeting C# 7.3.

In that case, I would prever `while (sr.ReadLine() is string line)` in this case.

Reply:
We should update to latest, I think.

Reply:
I prefer to be explicit. `latest` means different things for different builds of Visual Studio.

Reply:
That's fine. Any reason not to support C# 9 then?

Reply:
C# 9.0 is the default only for .NET 5.0 ([language defaults](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#defaults)).

It must be done carefully to support .NET Framework and .NET Standard 2.0 (Visual Studio will default to C# 7.3) because some features depend on types that are only on .NET Core 3.x/.NET Standard 2.1/.NET 5.0, but it can be done.